publishers/community/generic.py

```diff
@@ -163,3 +164,46 @@ def _recursive_enumerate_fields(structure, output_reference, path=''):
     _recursive_enumerate_fields(publication, output)
 
     return OrderedDict(sorted(output.items()))
+
+
+@Register
+def bubble_fields(alert, publication):
+    """This publisher bubbles all requested fields to the top level and deletes everything else.
+
+    It uses the context to determine which fields to keep. Example:
+
+    context={
+      'bubble_fields': [ 'field1', 'field2', 'field3' ]
+    }
+
+    "bubble_fields" should be an array of strings that are exact matches to the field names.
+
+    The algorithm deeply searches the publication for any dict key that exactly matches one of the
+    given fields. It then takes the contents of that field and bubbles them up to the top level.
+    It discovers ALL values matching each field, so if a field is returned multiple times, the
+    resulting top level field will be an array. In the special case where exactly one entry is
+    returned for a bubble_field, the value will instead be equal to that value (instead of an
+    array with 1 element being that value). In the special case when no entries are returned for
+    an extract_field, the value will be None.
+
+    Aside from the bubbled fields, this publisher throws away everything else in the original
+    publication.
+
+    NOTE: It is possible for bubbled fields to continue to contain nested dicts, so do not assume
+          this publisher will result in a flat dictionary publication.
+    """
+
+    new_publication = {}
+    for bubble_field in alert.context.get('bubble_fields', []):
+        extractions = get_keys(publication, bubble_field)
+
+        if len(extractions) > 1:
```

Comment:
I don't think it is necessary to check the length of `extractions`. In current logic, if the length of extractions larger than 1, the value will be a list. If the length is 1, the value will be a string. Is it what you want?

Reply:
Yeah that's what I wanted, since this is more of a function to generate user-facing data, instead of data that's being used by more code. I don't have a _strong_ opinion on this, since I think a 1 element array is not all that different.

I just don't like how PagerDuty renders 1 element arrays, since it shows up as:

```
[
  "element_here\n with newlines\noh no"
]
```

Versus the way I do it it'd show up as 

```
element here
with newlines 
oh no
```

Reply:
That being said, it's easy for me to have another PagerDuty-specific publisher that transforms arrays into cleaner PD-readable strings.

Reply:
I removed this logic and moved it into a new publisher.